### ISSUE TITLE
Volumes mounted as root in OpenShift, non-privileged containers can't write to them

### ISSUE THREAD

**Issue by fabianvf** [rook/rook#1921]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
When a pod consumes a PVC and the volume is mounted, the volume is owned by `root:root` and has the permissions set to `drwxr-x---`. In a non-privileged pod this makes it impossible to use the mounted volume.

**Expected behavior:**
The volume is mounted in a way that is consumable by the application, probably with the proper user and group as owners. This would put it in line with the documented behavior of external ceph volumes: https://docs.openshift.com/enterprise/3.1/install_config/persistent_storage/persistent_storage_ceph_rbd.html#ceph-volume-security

**How to reproduce it (minimal and precise):**

On an OpenShift cluster with rook deployed as described in the docs (and the example rook block storage class created), create a PersistentVolume claim, and a pod that mounts the volume. `exec` or `rsh` into the pod and attempt to write to the mounted volume.

```bash
$ cat example.yaml
---
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: ceph-pvc
spec:
  accessModes:
  - ReadWriteOnce
  storageClassName: rook-ceph-block
  resources:
    requests:
      storage: 1Gi
---
apiVersion: v1
kind: Pod
metadata:
  name: mount-test
spec:
  containers:
  - image: busybox
    command:
      - sleep
      - "3600"
    imagePullPolicy: IfNotPresent
    name: busybox
    volumeMounts:
    - mountPath: /ceph
      name: ceph
  restartPolicy: Always
  volumes:
    - name: ceph
      persistentVolumeClaim:
        claimName: ceph-pvc
$ oc create -f example.yaml
persistentvolumeclaim "ceph-pvc" created
pod "mount-test" created
$ #wait for pod to be ready
$ oc exec mount-test -- ls -lah /ceph
total 0
drwxr-x---    2 root     root           6 Jul 20 17:56 .
drwxr-xr-x    1 root     root          40 Jul 20 17:56 ..
$ oc exec mount-test -- touch /ceph/test
touch: /ceph/test: Permission denied
command terminated with exit code 1
```
**Environment**:
* OS (e.g. from /etc/os-release):
```
NAME=Fedora
VERSION="27 (Atomic Host)"
ID=fedora
VERSION_ID=27
PRETTY_NAME="Fedora 27 (Atomic Host)"
ANSI_COLOR="0;34"
CPE_NAME="cpe:/o:fedoraproject:fedora:27"
HOME_URL="https://fedoraproject.org/"
SUPPORT_URL="https://fedoraproject.org/wiki/Communicating_and_getting_help"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=27
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=27
PRIVACY_POLICY_URL="https://fedoraproject.org/wiki/Legal:PrivacyPolicy"
VARIANT="Atomic Host"
VARIANT_ID=atomic.host
```

* Kernel (e.g. `uname -a`):
```
Linux master.example.org 4.15.17-300.fc27.x86_64 #1 SMP Thu Apr 12 18:19:17 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```
* Cloud provider or hardware configuration:

Running on a libvirt VM with 16 CPUs (Intel Haswell, no TSX), 8GB RAM, 3 ~40GB virtual disks.
(Vagrant file here: https://github.com/fabianvf/home-cluster/blob/master/Vagrantfile)

* Rook version (use `rook version` inside of a Rook Pod):
```
rook: v0.7.0-284.g863c10f
```
* Kubernetes version (use `kubectl version`):
(using oc version instead)
```
oc v3.9.0+71543b2-33
kubernetes v1.9.1+a0ce1bc657
features: Basic-Auth GSSAPI Kerberos SPNEGO

Server https://master.example.org:8443
openshift v3.9.0+71543b2-33
kubernetes v1.9.1+a0ce1bc657
```
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift):
OpenShift
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/Rook/master/toolbox.html)):
```
HEALTH_OK
```

**Comment by fabianvf** [comment]
ping @travisn

**Comment by aland-zhang** [comment]
please add this lines in pod :
restartPolicy: Always
schedulerName: default-scheduler
securityContext: 
  runAsUser: 0

**Comment by travisn** [comment]
Note that this is a doc bug now given the investigation mentioned [here](https://github.com/rook/rook/pull/1930#issuecomment-411513655) about mount propagation issues on openshift when running in a container. Thanks for the investigation @fabianvf!

**Comment by travisn** [comment]
@fabianvf want to add your notes here on the openshift configuration until we can get the docs updated?

**Comment by fabianvf** [comment]
Yeah, sorry for the radio silence. The issue is pretty narrow, if you are running OpenShift on a normal OS then you should have no issues, but if you're running OpenShift (and maybe Kubernetes?) on atomic (or I assume CoreOS) based distributions, or running the node process in a container, then all the filesystem operations for flexvolumes will be performed inside the container and not propagate to the host.
